File: POOVeiculos/C#/Moto.cs
using System;

namespace POOVeiculos {
    class Moto : Veiculo {
        public override void engatar(char r) {
            if(Char.ToUpper(r) == 'R') {
                Console.WriteLine("Moto não tem ré! Empurre a moto!");
            }
            else {
                Console.WriteLine($"Não é possível engatar essa marcha no {this.GetType().Name.ToLower()} {this.modelo}.");
            }
        }

        public Moto(string modelo, string tipo, float potencia, int ano) : base(modelo, tipo, potencia, ano) {
        
        }
    }
}

File: POOVeiculos/Java/Moto.java
package pooveiculos;

public class Moto extends Veiculo {
    @Override
    public void engatar(char r) {
        if(Character.toUpperCase(r) == 'R') {
            System.out.println("Moto não tem ré! Empurre a moto!");
        }
        else {
            System.out.println("Não é possível engatar essa marcha no " + this.getClass().getSimpleName().toLowerCase() + " " + this.modelo + ".");
        }
    }
    
    public Moto(String modelo, String tipo, float potencia, int ano) {
        super(modelo, tipo, potencia, ano);
    }
}
